### ISSUE TITLE
Tests of some "difficult" words in Jarai [jra]

### ISSUE THREAD

**Issue by livingfield** [silnrsi/font-busra#4]
Line 2, 3, 4: on macOS Sonoma in both LibreOffice 7.6 and Microsoft Word, The register shifter 17c9 is rendered too far to the right on some words.

Line 5 and 6: In Microsoft Word, The register shifter 17ca is missing completely when it follows two coeng consonants.
[Busra Test B.docx](https://github.com/silnrsi/font-busra/files/13739333/Busra.Test.B.docx)
[Busra Test B.odt](https://github.com/silnrsi/font-busra/files/13739334/Busra.Test.B.odt)

**Comment by mhosken** [comment]
Please submit pdfs or images and plain text, and perhaps images/pdfs of
what you are wanting it to look like. Single words are sufficient.
TIA.

**Comment by livingfield** [comment]
[Busra Test B LibreOffice.pdf](https://github.com/silnrsi/font-busra/files/13775098/Busra.Test.B.LibreOffice.pdf)
[Busra Test B MS Word.pdf](https://github.com/silnrsi/font-busra/files/13775101/Busra.Test.B.MS.Word.pdf)

**Comment by mhosken** [comment]
fixed. I hope.

**Comment by livingfield** [comment]
No. I don't think any of these are fixed.

Here are a few problematic words in Microsoft Word 365 on Windows 11 Arm:
<img width="1063" alt="Image" src="https://github.com/user-attachments/assets/00443119-9d07-485c-9842-0037800924f0" />

Most of the problems also occur in Libreoffice on Windows.
<img width="1253" alt="Image" src="https://github.com/user-attachments/assets/57a36405-729e-4254-b4f1-3c00c23fed8a" />

[Busra.Dec2024WinLibreOffice.pdf](https://github.com/user-attachments/files/18462843/Busra.Dec2024WinLibreOffice.pdf)
[Busra.Dec2024inWinWord.pdf](https://github.com/user-attachments/files/18462844/Busra.Dec2024inWinWord.pdf)
[Busra.Dec2024inWinWord.docx](https://github.com/user-attachments/files/18462842/Busra.Dec2024inWinWord.docx)
[Busra.Dec2024LibreOffice.odt](https://github.com/user-attachments/files/18462845/Busra.Dec2024LibreOffice.odt)

**Comment by mhosken** [comment]
Sorry. I think you jumped the gun in testing. The bug is fixed and checked
in but not released yet. If you are testing a pre-release version, then
that is different. You should see the improvement in the next release.

Btw there is no need to give lots of contexts in which a font fails unless
the results differ between contexts.

**Comment by mhosken** [comment]
Regarding 5 & 6 in the Word case. I am assuming that if you printed that text, the -u (downshifted shifter) would reappear and that Word is clipping the shown text on the screen. There are problems with increasing that box for such a complex case in that it will increase the default linespacing which you probably don't want.

**Comment by livingfield** [comment]
> Regarding 5 & 6 in the Word case. I am assuming that if you printed that text, the -u (downshifted shifter) would reappear and that Word is clipping the shown text on the screen.

Indeed, it prints OK, and the PDF is OK. The problem is only on-screen.

**Comment by jvgaultney** [comment]
We may be able to improve 5 & 6 on screen by increasing usWinDescent. That will not affect line spacing except in really old Win apps, but would hopefully eliminate the clipping. I've now increased it slightly to include the full form of the -u even when it's the third glyph attached below. I don't know if that will help Word or not.

**Comment by livingfield** [comment]
Pages on macOS looks just like LibreOffice on Windows, though 2 and 3 is even further to the right. Hopefully the fixes that work on Windows will also work on macOS native apps.

Adobe Indesign 2025 in Windows renders the same as LibreOffice and Word.

**Comment by MakaraSok** [comment]
For 2 and 3, the shifter should stay on the coeng side per what is shown in a glossary of Khmerization. See below:

![Image](https://github.com/user-attachments/assets/47a6140c-e581-4019-a026-ccae71ef7b08)

Source: https://sala.moeys.gov.kh/kh/library/00004483 (page 9)

**Comment by viverechristus** [comment]
Thanks for pointing this out, Makara. What about the fourth item? This is another case where our team thought the shifter should be directly over the main/first consonant rather than on the coeung side -- but now I'm wondering whether we were mistaken about that, too.

**Comment by viverechristus** [comment]
> For 2 and 3, the shifter should stay on the coeng side per what is shown in a glossary of Khmerization.

I wonder about example line 2. In the example @MakaraSok  gives, the shifter applies directly to the coeung-ប, so it makes sense that it should sit over to the right (this corresponds to 3 in @livingfield 's table). But in line 2 of the table, the shifter applies to the base consonant, not the coeung, changing the pronunciation of ប. This is a situation that I suspect never occurs in Khmer, but ប+coeung automatically gives the ប a pronunciation of ប៉ (for example, ប្រាក់).
